### PR TITLE
Block Editor: Update the Convert to Blocks dialog

### PR DESCRIPTION
**Pull request by Copons**
#### Changes proposed in this Pull Request

* Update the copy of the Convert to Blocks dialog.
* Update the modal to use the Core component.
* Add a new persistent `ui.convertToBlocksDialog` state that keeps track of dismissed Convert to Blocks dialogs, in order to only show it once per post.

| Before | After |
| - | - |
| Image 2020-05-27 at 9 29 53 AM | Screenshot 2020-06-02 at 11 25 02 |

#### Testing instructions

* On a site using the Classic Editor, create a new post and write some content.
* Switch over to Gutenberg (Gutenframe).
* Make sure the "Convert to Blocks" notice shows up.
* Dismiss it (don't convert to blocks!) and reload the editor.
* Make sure the notice doesn't show up anymore.
* Switch back to Classic editor, and create another post with some content.
* Switch to Gutenberg, and make sure the notice shows up again.

Note: after switching to Gutenberg, adding **any** block content (including an empty Paragraph block) will automagically disable the notice as well.

Fixes https://github.com/Automattic/dotcom-manage/issues/87

### CODE REVIEW

File: client/components/convert-to-blocks/index.jsx [hunk 1]
```diff
@@ -1,62 +1,68 @@
...
-		);
-	}
+	return (
+		<Dialog
+			additionalClassNames="editor__gutenberg-convert-blocks-dialog"
+			buttons={ buttons }
+			isVisible={ showDialog }
+			onClose={ onClose }
+		>
+			<h1>{ translate( 'Convert to blocks?' ) }</h1>
```
Comment (a8ci18n) [inline client/components/convert-to-blocks/index.jsx:60]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **2** times:
`translate( 'Convert to blocks' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/convert-to-blocks-dialog)

**Comment by michelleweber** [comment]
Can we add some more info? I feel like we need to explain a bit more about the difference between "convert to blocks" and "add classic block" -- for someone not used to the block editor they both sound like "make some blocks."

Convert to blocks?
This post contains content you created with the classic editor. You can convert this content into blocks -- each paragraph and item will be its own block, that you can edit and move around -- or add a Classic block to simulate the classic editor.

File: client/components/convert-to-blocks/index.jsx [hunk 2]
```diff
@@ -1,62 +1,68 @@
...
-				<p>
-					{ translate(
-						'This post contains content you created using the older editor. For the best editing experience, we recommend converting this content to blocks.'
-					) }
-				</p>
-			</Dialog>
-		);
-	}
+	return (
+		<Dialog
```
Comment (scruffian) [inline client/components/convert-to-blocks/index.jsx:54]: What do you think about converting this to use `Modal` from Gutenberg?

Comment (Copons) [reply client/components/convert-to-blocks/index.jsx:54]: I've tried but it's missing lots of CSS. Maybe I'm missing something. 🤔

**Comment by Copons** [comment]
Copy updated! Thanks @michelleweber!

Screenshot 2020-06-01 at 19 09 57

**Comment by michelleweber** [comment]
Thanks! Let's lose the comma after "block," it's been bugging me and it's not needed.

Also: would it be helpful to try and trim a little to get rid of the last single word on the last line, or does it vary with screen size so it doesn't matter so much?

**Comment by Copons** [comment]
Updated the dialog to use the Core `Modal` component:

Screenshot 2020-06-02 at 11 25 02

Also @michelleweber I've removed the comma after "block".
And you guessed it right, the modal is responsive (both the previous Calypso version and this new Core one). 🙂

File: client/components/convert-to-blocks/index.jsx [hunk 3]
```diff
@@ -1,62 +1,67 @@
...
-					) }
-				</p>
-			</Dialog>
-		);
-	}
+	return (
+		<Modal
+			onRequestClose={ onClose( 'cancel' ) }
+			shouldCloseOnClickOutside={ false }
+			title={ translate( 'Convert to blocks?' ) }
```
Comment (a8ci18n) [inline client/components/convert-to-blocks/index.jsx:50]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **2** times:
`translate( 'Convert to blocks' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/convert-to-blocks-dialog)

File: client/state/selectors/test/is-convert-to-blocks-dialog-dismissed.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+		const isDismissed = isConvertToBlocksDialogDismissed( state, 12345678, 1 );
+		expect( isDismissed ).toBe( true );
+	} );
+
+	test( 'should return false if the dialog has not been dismissed for a given post', () => {
+		const isDismissed = isConvertToBlocksDialogDismissed( state, 12345678, 2 );
+		expect( isDismissed ).toBe( false );
+	} );
+
+	test( 'should return false if the site and is not in state', () => {
```
Comment (glendaviesnz) [inline client/state/selectors/test/is-convert-to-blocks-dialog-dismissed.js:27]: should this be `should return false if the site is not in state` ?

**Comment by glendaviesnz** [comment]
A rebase fixed the failing desktop test

**Comment by Copons** [comment]
I've moved this back to WIP because the new `ui.convertToBlocksDialog` state should live in the upcoming modularised `editor` state introduced in #43388.
Once that PR is merged, I'll rebase this and move `convertToBlocksDialog` over to `editor`.

**Comment by Copons** [comment]
In 38f8495 I've moved the new `convertToBlocksDialog` state from the `ui` root to the new modularized `editor`.

**Comment by creativecoder** [comment]
@Copons Even though we're discussing reworking this prompt to "Convert to Blocks" (pb5gDS-DR-p2), I think we should go ahead with this change, as it's a nice improvement, and this potentially gets us ready for the deprecation sooner.

I added the "String Freeze" label to queue up the new strings for translation--I think we should wait for those before merging to prevent non-English users from seeing English text before it's translated.

**Comment by creativecoder** [comment]
I've tested this change, and it works as expected:

- Clicking "Convert to Blocks" converts the content to blocks, modal does not appear on page reload
- Clicking "Stick with the Classic Block" or the X leaves content in the classic block, modal does not appear on page reload

One quirk is that in Firefox (Developer Edition), hovering over the X causes scrollbars to appear for the modal.

Screen Shot 2020-07-01 at 17 50 14

**Comment by Copons** [comment]
> One quirk is that in Firefox (Developer Edition), hovering over the X causes scrollbars to appear for the modal.

Oh, good catch, and it happens on Firefox "standard" and Chrome too.
Those scrollbars are caused by the Close button's tooltip being incorrectly positioned far below and to the right.

I believe it to be a Core bug.
We could fix by hiding the modal overflow, but I've chosen to remove the Close button instead.
It's not super useful anyway, and we really want folks to take a decision here, as opposed to just dismiss the modal (which makes the choice of using the Classic block in the user's stead)

**Comment by glendaviesnz** [comment]
Decision has been made to remove this modal in favour of a an extra block toolbar option so closing for now.

**Pull request closed without merging**